src/main/java/com/company/demoapplication/ImageProcessor.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT-0
+ */
+
+package com.company.demoapplication;
+
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.PutObjectResult;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClientBuilder;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+import com.amazonaws.util.IOUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+
+class ImageProcessor {
+
+    private static final String bwFolder = "bw-images/";
+    private static final String brightenFolder = "brighten-images/";
+
+    void run() {
+        try {
+            List<Message> messages = extractTasks(Main.sqsQueueUrl);
+
+            Main.logger().debug("Received " + messages.size() + " messages");
+
+            for (Message message : messages) {
+                Main.logger().debug("Processing message...");
+
+                BWImageProcessor bwip = new BWImageProcessor();
+                BrightenImageProcessor bip = new BrightenImageProcessor();
+
+                String imageKey;
+                try {
+                    imageKey = SqsMessage.deserialize(message.getBody()).imageKey;
+                } catch (RuntimeException e) {
+                    Main.logger().debug("Could not parse: " + message.getBody());
+                    continue;
+                }
+
+                String imageName = getNameFromKey(imageKey);
+                Main.logger().debug("Image name: " + imageName);
+                File outputFile = File.createTempFile(Instant.now().toString() + imageName, null);
+                IOUtils.copy(Main.s3Client().getObject(Main.bucketName, imageKey).getObjectContent(), new FileOutputStream(outputFile, false));
+
+                bwip.monochromeAndUpload(outputFile, imageName);
+                bip.brightenAndUpload(outputFile, imageName);
+
+                deleteFile(outputFile);
+                sqsClient.deleteMessage(Main.sqsQueueUrl, message.getReceiptHandle());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is using a single API operation, but a batch API operation exists. You could help improve efficiency by using the batch operation instead. The batch operation is deleteMessageBatch.